### PR TITLE
Update

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Update with ROS
Fix sending command when speeds are udpated.

### CODE REVIEW

File: src/teleop_twist_keyboard.py [hunk 1]
```diff
@@ -90,31 +184,29 @@ def vels(speed,turn):
 				speed = speed * speedBindings[key][0]
 				turn = turn * speedBindings[key][1]
 
-				print(vels(speed,turn))
+				print(vels(speed, turn))
 				if (status == 14):
 					print(msg)
 				status = (status + 1) % 15
+				continue
```
Comment (ar13pit) [inline src/teleop_twist_keyboard.py:190]: This will never let the publisher thread execute `update`

File: src/teleop_twist_keyboard.py [hunk 2]
```diff
@@ -53,35 +55,127 @@
...
+	repeat = rospy.get_param("~repeat_rate", 0.0)
+	key_timeout = rospy.get_param("~key_timeout", 0.0)
+	if key_timeout == 0.0:
+		key_timeout = None
+
+	pub_thread = PublishThread(repeat)
+
 	x = 0
 	y = 0
+	z = 0
```
Comment (ar13pit) [inline src/teleop_twist_keyboard.py]: Unused variable

File: src/teleop_twist_keyboard.py [hunk 3]
```diff
@@ -90,31 +184,29 @@ def vels(speed,turn):
...
 				status = (status + 1) % 15
+				continue
 			else:
+				# Skip updating cmd_vel if key timeout and robot already
+				# stopped.
+				if key == '' and x == 0 and y == 0 and th == 0:
+					continue
 				x = 0
 				y = 0
+				z = 0
```
Comment (ar13pit) [inline src/teleop_twist_keyboard.py]: Unused variable. Rather set speed and turn to `0` before publishing

File: src/teleop_twist_keyboard.py [hunk 4]
```diff
@@ -90,31 +184,29 @@ def vels(speed,turn):
...
-				print(vels(speed,turn))
+				print(vels(speed, turn))
 				if (status == 14):
 					print(msg)
 				status = (status + 1) % 15
+				continue
 			else:
+				# Skip updating cmd_vel if key timeout and robot already
+				# stopped.
+				if key == '' and x == 0 and y == 0 and th == 0:
```
Comment (ar13pit) [inline src/teleop_twist_keyboard.py:194]: The check of `x`, `y`, `th` seems incorrect. I think `pub_thread.x`, `pub_thread.y` and `pub_thread.th` should be checked instead

**Review by ar13pit — CHANGES_REQUESTED** [review]

File: src/teleop_twist_keyboard.py (hunk 1 above)
Comment (MatthijsBurgh) [reply src/teleop_twist_keyboard.py:190]: Yes, that is the goal. You shouldn't send a cmd message if you update the speed

File: src/teleop_twist_keyboard.py (hunk 4 above)
Comment (MatthijsBurgh) [reply src/teleop_twist_keyboard.py:194]: I just copied the code from the original ROS repo. But I think these will always be the same when ending up at this check.

**Review by ar13pit — APPROVED** [review]

**Pull request merged**